packages/cli/src/cli/autoformat.rs

```diff
@@ -207,24 +206,20 @@ fn indentation_for(file_or_dir: impl AsRef<Path>) -> Result<IndentOptions> {
     ))
 }
 
-fn collect_rs_files(folder: &Path, files: &mut Vec<PathBuf>) {
-    let Ok(folder) = folder.read_dir() else {
+fn collect_rs_files(folder: &impl AsRef<Path>, files: &mut Vec<PathBuf>) {
+    let Ok(folder) = folder.as_ref().read_dir() else {
         return;
     };
-
     // load the gitignore
-
     for entry in folder {
         let Ok(entry) = entry else {
             continue;
         };
-
         let path = entry.path();
-
         if path.is_dir() {
             collect_rs_files(&path, files);
+            return;
```

Comment:
If we return here, I think it will only read the first directory in this structure:
- Folder 1 (collect_rs_files is called from this path)
  - Folder 2 (It reaches this folder)
  - Folder 3 (collect_rs_files so it never reaches this folder)